Subject: sched, rt: Provide means of disabling cross-cpu bandwidth sharing
From: Peter Zijlstra <a.p.zijlstra@chello.nl>
Date: Thu Oct 06 22:39:14 CEST 2011

Normally the RT bandwidth scheme will share bandwidth across the
entire root_domain. However sometimes its convenient to disable this
sharing for debug purposes. Provide a simple feature switch to this
end.

Signed-off-by: Peter Zijlstra <a.p.zijlstra@chello.nl>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
---
 kernel/sched_features.h |    1 +
 kernel/sched_rt.c       |    3 +++
 2 files changed, 4 insertions(+)

Index: linux-3.2/kernel/sched_features.h
===================================================================
--- linux-3.2.orig/kernel/sched_features.h
+++ linux-3.2/kernel/sched_features.h
@@ -67,3 +67,4 @@ SCHED_FEAT(NONTASK_POWER, 1)
 SCHED_FEAT(TTWU_QUEUE, 1)
 
 SCHED_FEAT(FORCE_SD_OVERLAP, 0)
+SCHED_FEAT(RT_RUNTIME_SHARE, 1)
Index: linux-3.2/kernel/sched_rt.c
===================================================================
--- linux-3.2.orig/kernel/sched_rt.c
+++ linux-3.2/kernel/sched_rt.c
@@ -560,6 +560,9 @@ static int balance_runtime(struct rt_rq 
 {
 	int more = 0;
 
+	if (!sched_feat(RT_RUNTIME_SHARE))
+		return more;
+
 	if (rt_rq->rt_time > rt_rq->rt_runtime) {
 		raw_spin_unlock(&rt_rq->rt_runtime_lock);
 		more = do_balance_runtime(rt_rq);
